### ISSUE TITLE
torch._C._LinAlgError: linalg.svd: The algorithm failed to converge because the input matrix is ill-conditioned or has too many repeated singular values (error code: 1).

### ISSUE THREAD

**Issue by bruno686** [tvergara/tracr-injection#1]
When I run 'python main.py --task dyck --algorithm_loss true --seed 1' , there is a bug alert. I try different seed, but that doesn't work.

**Comment by tvergara** [comment]
I now uploaded a fix! The problem was that `max_sequence_length` was being set to less than the size of the data seen in training, so an example like `"is this balanced '()]{}' ?\nYes"` was being cut down to is this balanced '()]`, and therefore the supervised loss for the example was being not correctly computed (as we look to generate the 'yes' token). For now I added a hacky fix in the `main.py` to select the `max_sequence_length` based on the task being selected, rather than being a specific hyperparam.
